internal/api/middleware.go

```diff
@@ -53,3 +54,54 @@ func (s *Server) AuthMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
 		return next(c)
 	}
 }
+
+// VaultAuthMiddleware verifies JWT tokens and ensures users can only access their own vaults
+func (s *Server) VaultAuthMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		// Get token from header
+		authHeader := c.Request().Header.Get("Authorization")
+		if authHeader == "" {
+			return c.JSON(http.StatusUnauthorized, map[string]string{
+				"error": "Missing authorization header",
+			})
```

Comment:
```suggestion
			return c.JSON(http.StatusUnauthorized, NewErrorResponse("Missing authorization header"))
```

Reply:
can you change  all the other places that we use `map[string]string`, to `NewErrorResponse("error message")`